bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/internal/SWTFontProvider.java

```diff
@@ -105,10 +105,14 @@ public static void disposeFontRegistry(Device device) {
 		}
 	}
 
+	private static final String SWT_FONT_REGISTRY = "swt.fontRegistry";
+
 	private static SWTFontRegistry newFontRegistry(Device device) {
-		if (device instanceof Display display && display.isRescalingAtRuntime()) {
-			return new ScalingSWTFontRegistry(device);
+		if ("legacy".equals(System.getProperty(SWT_FONT_REGISTRY)) && device instanceof Display display
```

Comment:
Just in case, compare ignoring case.

```suggestion
		if ("legacy".equalsIgnoreCase(System.getProperty(SWT_FONT_REGISTRY)) && device instanceof Display display
```